core/src/main/java/org/apache/druid/data/input/impl/InputSourceSecurityConfig.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.data.input.impl;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import org.apache.druid.java.util.common.StringUtils;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.net.URI;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+public class InputSourceSecurityConfig
+{
+
+  public static InputSourceSecurityConfig ALLOW_ALL = new InputSourceSecurityConfig(null, null);
+  @JsonProperty
+  private final List<URI> allowPrefixList;
+  @JsonProperty
+  private final List<URI> denyPrefixList;
+
+  @JsonCreator
+  public InputSourceSecurityConfig(
+      @JsonProperty("allowPrefixList") @Nullable List<URI> allowPrefixList,
+      @JsonProperty("denyPrefixList") @Nullable List<URI> denyPrefixList
+  )
+  {
+    this.allowPrefixList = allowPrefixList;
+    this.denyPrefixList = denyPrefixList;
+    Preconditions.checkArgument(
+        this.denyPrefixList == null || this.allowPrefixList == null,
+        "Can only use one of allowList or blackList"
+    );
+  }
+
+  public List<URI> getallowPrefixList()
+  {
+    return allowPrefixList;
+  }
+
+  public List<URI> getdenyPrefixList()
+  {
+    return denyPrefixList;
+  }
+
+  private static boolean matchesAny(List<URI> domains, URI uri)
+  {
+    URI toMatch = uri.normalize();
+    for (URI prefix : domains) {
+      return toMatch.toString().startsWith(prefix.toString());
```

Comment:
should return only if there is a match.

Reply:
fixed. thanks for catching it up. Added a test as well.